src/typescript/typescript.ts

```diff
@@ -22,6 +23,158 @@ import {
 } from "../typescript";
 import { deepMerge } from "../util";
 
+export class TsJestBabelConfig {
+  public static default() {
+    return new TsJestBabelConfig(true);
+  }
+
+  public static fromFile(filePath: string) {
+    return new TsJestBabelConfig(filePath);
+  }
+
+  public static custom(config: Record<string, any>) {
+    return new TsJestBabelConfig(config);
+  }
+
+  private constructor(
+    private readonly config: boolean | string | Record<string, any>
+  ) {}
+
+  /**
+   * @jsii ignore
+   * @internal
+   */
+  public toJSON(): boolean | string | Record<string, any> {
+    return this.config;
+  }
+}
+
+export class TsJestDiagnostics {
+  public static default() {
```

Comment:
As per the docs: "Disabling/enabling
By default all diagnostics are enabled. This is the same as setting the diagnostics option to true. To disable all diagnostics, set diagnostics to false. This might lead to slightly better performance, especially if you're not using Jest's cache."

```suggestion
/**
 * Enable all diagnostics.
 */
  public static all() {
```